core/flyout.js

```diff
@@ -260,28 +329,46 @@ Blockly.Flyout.prototype.position = function() {
     // Hidden components will return null.
     return;
   }
-  var edgeWidth = this.width_ - this.CORNER_RADIUS;
-  if (this.RTL) {
+  var edgeWidth = this.horizontalLayout_ ? metrics.viewWidth : this.width_;
+  edgeWidth -= this.CORNER_RADIUS;
+  if (this.toolboxPosition_ == Blockly.TOOLBOX_AT_RIGHT) {
     edgeWidth *= -1;
   }
 
-  this.setBackgroundPath_(edgeWidth, metrics.viewHeight);
+  this.setBackgroundPath_(edgeWidth,
+      this.horizontalLayout_ ? this.height_ : metrics.viewHeight);
 
   var x = metrics.absoluteLeft;
-  if (this.RTL) {
+  if (this.toolboxPosition_ == Blockly.TOOLBOX_AT_RIGHT) {
     x += metrics.viewWidth;
     x -= this.width_;
   }
+
+  var y = metrics.absoluteTop;
+  if (this.toolboxPosition_ == Blockly.TOOLBOX_AT_BOTTOM) {
+    y += metrics.viewHeight;
+    y -= this.height_;
+  }
+
   this.svgGroup_.setAttribute('transform',
-      'translate(' + x + ',' + metrics.absoluteTop + ')');
+      'translate(' + x + ',' + y + ')');
 
-  // Record the height for Blockly.Flyout.getMetrics_.
-  this.height_ = metrics.viewHeight;
+  // Record the height for Blockly.Flyout.getMetrics_, or width if the layout is
+  // horizontal.
+  if (this.horizontalLayout_) {
+    this.width_ = metrics.viewWidth;
+  } else {
+    this.height_ = metrics.viewHeight;
+  }
 
   // Update the scrollbar (if one exists).
   if (this.scrollbar_) {
     this.scrollbar_.resize();
   }
+  // The blocks need to be visible in order to be laid out and measured
+  // correctly, but we don't want the flyout to show up until it's properly
+  // sized.  Opacity is set to zero in show().
+  this.svgGroup_.style.opacity = 1;
```

Comment:
Ah, I hadn't gotten to show yet.  What happens when you click on something that is transparent? I think want the event to go through it to the thing below, but will it?  My first thought would have been to use visiblity:hidden instead of opacity, but my thoughts on these things come from old internet times so might be outdated.